doc/specs/Configuration object for profiles.md

```diff
@@ -0,0 +1,125 @@
+pane is focused and which panes are unfocused. This change would grant us that feature.
+
+## Solution Design
+
+The implementation design for appearance config objects centers around the recent change where inheritance was added to the
+`TerminalSettings` class in the Terminal Settings Model - i.e. different `TerminalSettings` objects can inherit from each other.
+The reason for this change was that we did not want a settings reload to erase any overrides `TermControl` may have made
+to the settings during runtime. By instead passing a child of the `TerminalSettings` object to the control, we can change
+the parent of the child during a settings reload without the overrides being erased (since those overrides live in the child).
+
+The idea behind unfocused appearance configurations is similar. We will pass in another `TerminalSettings` object to the control,
+which is simply a child that already has some overrides in it. When the control gains or loses focus, it simply switches between
+the two settings objects appropriately.
+
+### Allowed parameters
+
+For now, these states are meant to be entirely appearance-based. So, not all parameters which can be
+defined in a `Profile` can be defined in this new object (for example, we do not want parameters which
+would cause a resize in this object.) Here is the list of parameters we will allow:
+
+- Anything regarding colors: `colorScheme`, `foreground`, `background`, `cursorColor` etc
+- Anything regarding background image: `path`, `opacity`, `alignment`, `stretchMode`
+- `cursorShape`
+
+We may wish to allow further parameters in these objects in the future (like `bellStyle`?). The addition
+of further parameters can be discussed in the future and is out of scope for this spec.
+
+### Inheritance
+
+The inheritance model can be thought of as an 'all-or-nothing' approach in the sense that the `unfocusedConfig` object
+is considered as a *single* setting instead of an object with many settings. We have chosen this model because it is cleaner
+and easier to understand than the alternative, where each setting within an `unfocusedConfig` object has a parent from which
+it obtains its value.
+
+Note that when `TerminalApp` initializes a control, it creates a `TerminalSettings` object for that profile and passes the
+control a child of that object (so that the control can store overrides in the child, as described earlier). If an unfocused
+config is defined in the profile (or in globals/profile defaults), then `TerminalApp` will create a *child of that child*,
+put the relevant overrides in it, and pass that into the control as well. Thus, the inheritance of any undefined parameters
+in the unfocused config will be as follows:
+
+1. The unfocused config specified in the profile (or in globals/profile defaults)
+2. Overrides made by term control
+3. The parent profile
+
+## UI/UX Design
+
+Users will be able to add a new setting to their profiles that will look like this:
+
+```
+"unfocusedConfig":

```

Comment:
Had we agreed on unfocusedConfig vs unfocusedAppearance? The second one ties our hands, but might be more obvious to the users.

Reply:
Okay! Updated